CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -33,6 +33,96 @@
 class CRM_Contact_BAO_Contact_Permission {
 
   /**
+   * Check which of the given contact IDs the logged in user
+   *   has permissions for the operation type according
+   *
+   * Caution: general permissions (like 'edit all contacts')
+   *
+   * @param array $contact_ids
+   *   Contact IDs.
+   * @param int|string $type the type of operation (view|edit)
+   *
+   * @see CRM_Contact_BAO_Contact_Permission::allow
+   *
+   * @return array
+   *    list of contact IDs the logged in user has the given permission for
+   */
+  public static function allowList($contact_ids, $type = CRM_Core_Permission::VIEW) {
+    $result_set = array();
+    if (empty($contact_ids)) {
+      // empty contact lists would cause trouble in the SQL. And be pointless.
+      return $result_set;
+    }
+    $contact_id_list = implode(',', $contact_ids);
+
+    // make sure the the general permissions are given
+    if (CRM_Core_Permission::check('edit all contacts')
+        || $type == CRM_Core_Permission::VIEW && CRM_Core_Permission::check('view all contacts')
+      ) {
+
+      // if the general permission is there, all good
+      if (CRM_Core_Permission::check('access deleted contacts')) {
+        // if user can access delted contacts -> fine
+        return $contact_ids;
+      }
+      else {
+        // if the user CANNOT access deleted contacts, these need to be filtered
+        $filter_query = "SELECT DISTINCT(id) FROM civicrm_contact WHERE id IN ($contact_id_list) AND is_deleted = 0";
+        $query = CRM_Core_DAO::executeQuery($filter_query);
+        while ($query->fetch()) {
+          $result_set[(int) $query->id] = TRUE;
+        }
+        return array_keys($result_set);
+      }
+    }
+
+    // get logged in user
+    $session   = CRM_Core_Session::singleton();
+    $contactID = (int) $session->get('userID');
```

Comment:
will do (found out it's ``::getLoggedInContactID()``, just in case anybody wonders)